core/store/store.go

```diff
@@ -46,200 +34,61 @@ type NotifyNewEthTx interface {
 // for keeping the application state in sync with the database.
 type Store struct {
 	*orm.ORM
-	Config            *orm.Config
-	Clock             utils.AfterNower
-	KeyStore          KeyStoreInterface
-	VRFKeyStore       *VRFKeyStore
-	TxManager         TxManager
-	GethClientWrapper GethClientWrapper
-	NotifyNewEthTx    NotifyNewEthTx
-	closeOnce         *sync.Once
-}
-
-type lazyRPCWrapper struct {
-	client      *rpc.Client
-	url         *url.URL
-	mutex       *sync.Mutex
-	initialized *abool.AtomicBool
-	limiter     *rate.Limiter
-}
-
-func newLazyRPCWrapper(urlString string, limiter *rate.Limiter) (eth.CallerSubscriber, error) {
-	parsed, err := url.ParseRequestURI(urlString)
-	if err != nil {
-		return nil, err
-	}
-	if parsed.Scheme != "ws" && parsed.Scheme != "wss" {
-		return nil, fmt.Errorf("ethereum url scheme must be websocket: %s", parsed.String())
-	}
-	return &lazyRPCWrapper{
-		url:         parsed,
-		mutex:       &sync.Mutex{},
-		initialized: abool.New(),
-		limiter:     limiter,
-	}, nil
-}
-
-// lazyDialInitializer initializes the Dial instance used to interact with
-// an ethereum node using the Double-checked locking optimization:
-// https://en.wikipedia.org/wiki/Double-checked_locking
-func (wrapper *lazyRPCWrapper) lazyDialInitializer() error {
-	if wrapper.initialized.IsSet() {
-		return nil
-	}
-
-	wrapper.mutex.Lock()
-	defer wrapper.mutex.Unlock()
-
-	if wrapper.client == nil {
-		client, err := rpc.Dial(wrapper.url.String())
-		if err != nil {
-			return err
-		}
-		wrapper.client = client
-		wrapper.initialized.Set()
-	}
-	return nil
-}
-
-// GethClient allows callers to access go-ethereum's ethclient through the
-// wrapper's rate limiting
-func (wrapper *lazyRPCWrapper) GethClient(callback func(gethClient eth.GethClient) error) error {
-	err := wrapper.lazyDialInitializer()
-	if err != nil {
-		return err
-	}
-	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
-	defer cancel()
-	logger.ErrorIf(wrapper.limiter.Wait(ctx))
-
-	client := gethClient.NewClient(wrapper.client)
-
-	return callback(client)
-}
-
-// RPCClient allows callers to access go-ethereum's rpcclient through the
-// wrapper's rate limiting
-func (wrapper *lazyRPCWrapper) RPCClient(callback func(rpcClient eth.RPCClient) error) error {
-	err := wrapper.lazyDialInitializer()
-	if err != nil {
-		return err
-	}
-	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
-	defer cancel()
-	logger.ErrorIf(wrapper.limiter.Wait(ctx))
-
-	return callback(wrapper.client)
+	Config         *orm.Config
+	Clock          utils.AfterNower
+	KeyStore       KeyStoreInterface
+	VRFKeyStore    *VRFKeyStore
+	TxManager      TxManager
+	EthClient      eth.Client
```

Comment:
It probably doesn't belong on the store.

Reply:
I eventually would like to get rid of the store, but curious about why you think it doesn't belong here in the meantime.

Reply:
I agree with Sam.
In my head, the store is the abstraction/interface for the persistence layer. 
It shouldn't do anything else. It shouldn't know about Geth!
It's the same reason why I think Config shouldn't be in the store.

Reply:
My reasoning is that, because we've historically used the TxManager as our EthClient, the new EthClient should live in the same place that the TxManager does.  With that said, I'm not convinced that either of them belong on the store, but again, I'd vote to push that into a separate PR.

Reply:
The store may be a bit of a catch-all bucket at this point, which is itself a justifiable reason to clean this stuff up. FWIW we justified the Ethereum client and the config as other persistence layers when we put them on here, just ones with different behaviors than a traditional DB. I still think that's a reasonable framing, but I'm open to separating it.